Add automation for running gatling nightly

gerlowskija opened the pull request:
Primary entrypoint is `run-benchmark-on-commits.sh`, which takes a
list of commits (or a "branch", "startCommit" and "endCommit")
and iterates over them, running Gatling benchmarks on each and storing
the results in a data directory.

Users who want to automate running benchmarks periodically (i.e.
nightly) may use the provided `periodic-benchmark.sh` script.  This isn't
as nice as having a full Jenkins setup, that's probably the "right" way to
do this sort of recurring automation.

But providing a simple path for folks to get up and running that doesn't
involve installing and running Jenkins seems like a nice 80/20 tradeoff
that lowers the barrier-to-entry a good deal.

epugh commented on scripts/gatling/lib/solr.sh:
```diff
@@ -0,0 +1,66 @@
+  fi
+
+  pushd $location
+    git clone git@github.com:apache/solr.git
```
```suggestion
    git clone https://github.com/apache/solr.git
```

epugh commented:
to get solr to start, i had to add `export SOLR_JETTY_HOST=0.0.0.0` or it never saw that solr started...

epugh commented:
also had to add `--force` to start command.

epugh commented on scripts/gatling/lib/solr.sh:
```diff
@@ -0,0 +1,66 @@
+}
+
+function solr_kill_all() {
+  ps -ef | grep solr | grep java | awk {'print $2'} | xargs kill -9
```
```suggestion
  ps -ef | grep solr | grep java | awk {'print $2'} | xargs -r kill -9
```

epugh replied:
if solr isn't running the then kill gets no args, and errors...  so add `-r`.

epugh commented on scripts/gatling/run-benchmark-on-commits.sh:
```diff
@@ -0,0 +1,102 @@
+    solr_build_package
+    package_dir="$(solr_get_package_directory)"
+    pushd $package_dir
+      export START_SOLR_OPTS=" -m 4g "
```
does this overwrite my own START_SOLR_OPTS ???

epugh commented on scripts/gatling/lib/solr.sh:
```diff
@@ -0,0 +1,66 @@
+# Start Solr (with custom ops pulled from START_SOLR_OPTS)
+# Assumes cwd of the Solr package root
+function solr_start() {
+  bin/solr start ${START_SOLR_OPTS:-}
```
```suggestion
  bin/solr start -c ${START_SOLR_OPTS:-}
```

epugh replied:
we are going to want to add some backcompat in solr 10 so -c doesn't raise an error I bet!

epugh commented on scripts/gatling/run-benchmark-on-commits.sh:
```diff
@@ -0,0 +1,102 @@
+    package_dir="$(solr_get_package_directory)"
+    pushd $package_dir
+      export START_SOLR_OPTS=" -m 4g "
+      solr_start
```
so, yeah, I wanted to acces solr from externally.  I was running solr on a Digital Ocean droplet, and wanted to log into solr and see how it was doing...

gerlowskija commented:
Alright, I think I've addressed all your comments above except for the `SOLR_JETTY_HOST` bit @epugh .  Happy to revise this more, but otherwise will aim to merge in the next day or two and we can iterate from there!  Thanks for the review!

epugh commented:
awesome.
